server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java

```diff
@@ -0,0 +1,318 @@
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AwaitsFix(bugUrl = "waiting upload flow rebase. tested on integration PR")
+    public void testFullClusterRestore() throws Exception {
+        int shardCount = randomIntBetween(1, 2);
+        int replicaCount = 0;
+        int dataNodeCount = shardCount;
+        int clusterManagerNodeCount = 1;
+
+        // Step - 1 index some data to generate files in remote directory
+        Map<String, Long> indexStats = initialTestSetup(shardCount, replicaCount, dataNodeCount, 1);
+        String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+
+        // Step - 2 Replace all nodes in the cluster with new nodes. This ensures new cluster state doesn't have previous index metadata
+        resetCluster(dataNodeCount, clusterManagerNodeCount);
+
+        String newClusterUUID = clusterService().state().metadata().clusterUUID();
+        assert !Objects.equals(newClusterUUID, prevClusterUUID) : "cluster restart not successful. cluster uuid is same";
+
+        // Step - 3 Trigger full cluster restore and validate
+        restoreAndValidate(prevClusterUUID, indexStats);
+    }
+
+    @AwaitsFix(bugUrl = "waiting upload flow rebase. tested on integration PR")
+    public void testFullClusterRestoreMultipleIndices() throws Exception {
+        int shardCount = randomIntBetween(1, 2);
+        int replicaCount = 0;
+        int dataNodeCount = shardCount + 1;
+        int clusterManagerNodeCount = 1;
+
+        // Step - 1 index some data to generate files in remote directory
+        Map<String, Long> indexStats = initialTestSetup(shardCount, replicaCount, dataNodeCount, clusterManagerNodeCount);
+
+        String secondIndexName = INDEX_NAME + "-2";
+        createIndex(secondIndexName, remoteStoreIndexSettings(replicaCount, shardCount + 1));
+        Map<String, Long> indexStats2 = indexData(1, false, secondIndexName);
+        assertEquals(shardCount + 1, getNumShards(secondIndexName).totalNumShards);
+        ensureGreen(secondIndexName);
+
+        String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+
+        // Step - 2 Replace all nodes in the cluster with new nodes. This ensures new cluster state doesn't have previous index metadata
+        resetCluster(dataNodeCount, clusterManagerNodeCount);
+
+        String newClusterUUID = clusterService().state().metadata().clusterUUID();
+        assert !Objects.equals(newClusterUUID, prevClusterUUID) : "cluster restart not successful. cluster uuid is same";
+
+        // Step - 3 Trigger full cluster restore
+        restoreAndValidate(prevClusterUUID, indexStats);
+        ensureGreen(secondIndexName);
+        verifyRestoredData(indexStats2, secondIndexName);
+    }
+
+    @AwaitsFix(bugUrl = "waiting upload flow rebase. tested on integration PR")
+    public void testFullClusterRestoreShardLimitReached() throws Exception {
```

Comment:
see if you can combine certain tests using the same setup. It might be expensive to setup cluster everytime for a simple exception test

Reply:
done. combined all validation failure related tests into one